apps/explorer/src/components/ownedobjects/views/UnderlyingObjectCard.tsx

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useGetDynamicFieldObject } from '@mysten/core';
+import { getObjectDisplay, getObjectFields, getObjectId } from '@mysten/sui.js';
+
+import { SyntaxHighlighter } from '~/components/SyntaxHighlighter';
+import DisplayBox from '~/components/displaybox/DisplayBox';
+import { Banner } from '~/ui/Banner';
+import { ObjectLink } from '~/ui/InternalLink';
+import { LoadingSpinner } from '~/ui/LoadingSpinner';
+import { Text } from '~/ui/Text';
+import {
+    parseImageURL,
+    parseObjectType,
+    extractName,
+} from '~/utils/objectUtils';
+import { trimStdLibPrefix } from '~/utils/stringUtils';
+
+interface UnderlyingObjectCardProps {
+    parentId: string;
+    name: {
+        type: string;
+        value?: string;
+    };
+}
+
+export function UnderlyingObjectCard({
+    parentId,
+    name,
+}: UnderlyingObjectCardProps) {
+    const { data, isLoading, isError, isFetched } = useGetDynamicFieldObject(
+        parentId,
+        name
+    );
+
+    if (isLoading) {
+        return (
+            <div className="mt-3 flex w-full justify-center pt-3">
+                <LoadingSpinner text="Loading data" />
+            </div>
+        );
+    }
+
+    if (isError || data.error || (isFetched && !data)) {
+        return (
+            <Banner variant="error" spacing="lg" fullWidth>
+                Failed to get field data for :{parentId}
+            </Banner>
+        );
+    }
+
+    const fieldsData = getObjectFields(data);
+    const display = getObjectDisplay(data);
+    const imgUrl = parseImageURL(display.data);
+
+    const objectType = parseObjectType(data);
+    const caption = extractName(display.data) || trimStdLibPrefix(objectType);
+
+    // Get content inside <> and split by , to get underlying object types
+    const underlyingObjectTypes = objectType
+        ?.slice(objectType?.indexOf('<') + 1, objectType.indexOf('>'))
+        .split(',');
+
+    // Split the first object type by :: and if array length is > 1 then it is a underlying object
+    const hasUnderlyingObject =
```

Comment:
If the intention here is to check that value is an object why not just do `typeof fieldsData.value === 'object'` seems easier and less confusing